### ISSUE TITLE
Unable to reproduce CityScape performance

### ISSUE THREAD

**Issue by Sara-Ahmed** [facebookresearch/capi#14]
Thanks a lot for the great work. I have a question regarding cityscapes. I am getting extremely low performance on linear evaluation.

I am using the torchvision dataset class using mode as gtFine and images are from "leftImg8bit".

Any idea why the linear evaluation would be that bad (around 15 MIoU) and KNN is actually better at around 25 MIoU

**Comment by TimDarcet** [comment]
Hi,
Hard to say from here. Linear being below knn hints at something being very wrong.
--> What `ignore_labels` do you use? The default in the codebase was set recently to `[0,255]` (for ADE20K) but for Cityscapes it should be `[255]`
--> To try debugging, can you get a confusion matrix of both knn and linear predictions?

**Comment by Sara-Ahmed** [comment]
I think it is more of what setting did you use for Cityscape. 

Let me tell you the exact settings  I am using
- Train: gtFine train split data
- Test: gtFine val split data
- ignore_labels [-1, 255]
- classes, using this guide, I set the label Id's to train_id, that is labels from 0 to 18 and -1 and 255 are ignored.

Are you using similar setting with input size 224?

    CityscapesClass = namedtuple(
        "CityscapesClass",
        ["name", "id", "train_id", "category", "category_id", "has_instances", "ignore_in_eval", "color"],
    )

    classes = [
        CityscapesClass("unlabeled", 0, 255, "void", 0, False, True, (0, 0, 0)),
        CityscapesClass("ego vehicle", 1, 255, "void", 0, False, True, (0, 0, 0)),
        CityscapesClass("rectification border", 2, 255, "void", 0, False, True, (0, 0, 0)),
        CityscapesClass("out of roi", 3, 255, "void", 0, False, True, (0, 0, 0)),
        CityscapesClass("static", 4, 255, "void", 0, False, True, (0, 0, 0)),
        CityscapesClass("dynamic", 5, 255, "void", 0, False, True, (111, 74, 0)),
        CityscapesClass("ground", 6, 255, "void", 0, False, True, (81, 0, 81)),
        CityscapesClass("road", 7, 0, "flat", 1, False, False, (128, 64, 128)),
        CityscapesClass("sidewalk", 8, 1, "flat", 1, False, False, (244, 35, 232)),
        CityscapesClass("parking", 9, 255, "flat", 1, False, True, (250, 170, 160)),
        CityscapesClass("rail track", 10, 255, "flat", 1, False, True, (230, 150, 140)),
        CityscapesClass("building", 11, 2, "construction", 2, False, False, (70, 70, 70)),
        CityscapesClass("wall", 12, 3, "construction", 2, False, False, (102, 102, 156)),
        CityscapesClass("fence", 13, 4, "construction", 2, False, False, (190, 153, 153)),
        CityscapesClass("guard rail", 14, 255, "construction", 2, False, True, (180, 165, 180)),
        CityscapesClass("bridge", 15, 255, "construction", 2, False, True, (150, 100, 100)),
        CityscapesClass("tunnel", 16, 255, "construction", 2, False, True, (150, 120, 90)),
        CityscapesClass("pole", 17, 5, "object", 3, False, False, (153, 153, 153)),
        CityscapesClass("polegroup", 18, 255, "object", 3, False, True, (153, 153, 153)),
        CityscapesClass("traffic light", 19, 6, "object", 3, False, False, (250, 170, 30)),
        CityscapesClass("traffic sign", 20, 7, "object", 3, False, False, (220, 220, 0)),
        CityscapesClass("vegetation", 21, 8, "nature", 4, False, False, (107, 142, 35)),
        CityscapesClass("terrain", 22, 9, "nature", 4, False, False, (152, 251, 152)),
        CityscapesClass("sky", 23, 10, "sky", 5, False, False, (70, 130, 180)),
        CityscapesClass("person", 24, 11, "human", 6, True, False, (220, 20, 60)),
        CityscapesClass("rider", 25, 12, "human", 6, True, False, (255, 0, 0)),
        CityscapesClass("car", 26, 13, "vehicle", 7, True, False, (0, 0, 142)),
        CityscapesClass("truck", 27, 14, "vehicle", 7, True, False, (0, 0, 70)),
        CityscapesClass("bus", 28, 15, "vehicle", 7, True, False, (0, 60, 100)),
        CityscapesClass("caravan", 29, 255, "vehicle", 7, True, True, (0, 0, 90)),
        CityscapesClass("trailer", 30, 255, "vehicle", 7, True, True, (0, 0, 110)),
        CityscapesClass("train", 31, 16, "vehicle", 7, True, False, (0, 80, 100)),
        CityscapesClass("motorcycle", 32, 17, "vehicle", 7, True, False, (0, 0, 230)),
        CityscapesClass("bicycle", 33, 18, "vehicle", 7, True, False, (119, 11, 32)),
        CityscapesClass("license plate", -1, -1, "vehicle", 7, False, True, (0, 0, 142)),
    ]

**Comment by TimDarcet** [comment]
For leftImg8bit/gtFine, I think it's correct but can't remember, I did the setup 4 years ago. Maybe @baldassarreFe can check, but that will probably be next week, as he's rushing for NeurIPS (and I'm not at Meta any more).

Other question to debug: Are you trying to eval a custom model, or a preexisting model? If you can try evaluating a model for which we have published numbers (eg capi), it will allow you to validate whether the eval is running correctly.

**Comment by Sara-Ahmed** [comment]
I am evaluating using the pre-trained weights of CAPI on INet-1K

After following the setup that I mentioned earlier. Here are the results I am getting
KNN = 29.5     Linear = 31.3

However, the reported numbers in the paper are
KNN = 35.6     Linear = 41.7

**Comment by TimDarcet** [comment]
I see, that's better.
IIRC I tried using the torchvision dataset and it did not give the same results as our internal loader for cityscapes. Let me try and discuss that with others to see what could cause the discrepancy

**Comment by TimDarcet** [comment]
Hey quick update, there's no obvious diff between our loader and the torchvision one, so we'll have to wait until next week to take a look with @baldassarreFe.
Sorry if you were rushing for neurips, there's not much more I can do

**Comment by TimDarcet** [comment]
Great to hear!
